lib/vsphere.rb

```diff
@@ -237,6 +247,22 @@ def execute options
 
   end
 
+  # Wait for the ip address to be assigned by VM before returning
+  def wait_for_ip(new_vm, wait, options)
+    puts "Waiting for IP address to be obtained by VM..."
+    poll_interval = 10
+    while wait > 0 && new_vm.guest_ip.nil?
+      debug( 'INFO', "Waiting for IP... #{wait} seconds remain" ) if options[:debug]
+      wait = wait - poll_interval
+      sleep poll_interval
+    end
+    if wait <= 0
```

Comment:
Probably super unlikely but if you are kicked out of this loop once wait == 0 and the new_vm is permuted to get an IP it would be possible to give a false error.  Your could instead check if the ip is still nil once you've left this loop `if new_vm.guest_ip.nil?`.  That would also protect you from a nil ip in your else statement.